### ISSUE TITLE
Client Name Length should be limited to 15 chars

### ISSUE THREAD

**Issue by jacksnodgrass** [Nyr/wireguard-install#114]
I created a client - racknerd2_mailbox which to me, describes my client.  Lets me look at it and know which server / client the config is for.  I copy that to my client and  do a 
wg-quick up racknerd2_mailbox 
and get
wg-quick: `racknerd2_mailbox' does not exist
after about 30 minutes of trying different things I find that the linux doesn't support interface names longer than 15 chars.... so racknerd2 works but not racknerd2_mailbox. 

It would be nice if the script that creates the .conf file limited the length to 15 chars.

**Comment by Nyr** [comment]
I will take this into consideration.

**Comment by stcrd** [comment]
I ran into the same issue. Seems like linux interface names can't be longer than 15 chars.

Here is the discussion on stackoverflow:
https://stackoverflow.com/questions/24932172/what-length-can-a-network-interface-name-have

**Comment by KaeTuuN** [comment]
@jacksnodgrass & @stcrd  I think you both mixed up two different things. Device has two meanings in this context.
First: The *NETWORK* device, which is only the logical Interface that wireguard creates for its VPN. This name is limited to **14** visible Characters and is called wg0 in Standard.
Second: The *Hardware* device using wireguard, for example a Server, Tablet, Laptop or Smartphone. This Device can have much longer names. When I'm right, this name is only used for the conf File name and as comment in the server conf to mark start and end of a Client. I just created one with 18 chars and worked without any Problems (Ubuntu Server 22.04 LTS).

Greetings
Kae

**Comment by jacksnodgrass** [comment]
@KaeTuuN - when you run the script, it asks for a client name... I gave it: racknerd2_mailbox and the script created a racknerd2_mailbox.conf file. The system ( fedora linux ) can't use that config / name because it trys to create an interface called racknerd2_mailbox and that is too long of an interface name and the command: "wg-quick up racknerd2_mailbox" fails to execute even though the racknerd2_mailbox.conf file exists and is in the expected location.  the wg-quick up racknerd2_mailbox command returns with: wg-quick: `racknerd2_mailbox' does not exist.

**Comment by KaeTuuN** [comment]
Ahhh... I see! So this looks like a Linux specific Problem. On Windows and Android (just tested), I can use 18 Chars long .conf Files without any problems. But you're right, since Linux uses the conf Filename as the Devicename this Problem occurs. I never ran into that because when moving/copying to other Linux Machines I rename the conf File to wg0.conf.

Workaround: Create the conf with the name you like on the WG Server and just rename it while copying to the Client Site to a shorter one. Then you have the correct names in the WG Server Conf and as a backup in the created conf Files on the Server.

**Comment by stcrd** [comment]
Yes, I was talking about Linux, too. I did that, just renamed the conf file on the client and it works.

Also it has nothing to do with visible/invisible characters. There is a const in the source with length of 16 bytes. And since all char arrays in C are null terminated, the limit is 15 chars.

**Comment by jacksnodgrass** [comment]
I understand the workaround..... the issue I have... not a huge / insurmountable one is just that when you make a client with a long name and copy it to the client....  you get a un-expected error about not being able to start wireguard because it can't find the looks-ok-to-me-and-the-script-generated-it-so-whats-the-problem device when you start it up.  i.e. the script created racknerd2_mailbox and linux goes to say: "wg-quick: `racknerd2_mailbox' does not exist"....  but it did exists... I see it... it's in the right place..... it just doesn't work.  IF I knew the work around then... I wouldn't have spent 30 minutes debugging the issue....  Also... I have 10 servers and different VPNs.... wg0 isn't near as descriptive as racknerd2_mailbox when I am looking at device names.  I don't like just using wg0, wg1, wg2 if I can use descriptive names.

**Comment by stcrd** [comment]
I totally agree with you. Linux error msg is not informative.

I am not a bash guru, maybe someone can make a PR with a simple length check or just include a notice about the 15 char limit.

**Comment by KaeTuuN** [comment]
Created #119 to alleviate the Problem in a first step.
